.github/workflows/test-on-push.yml

```diff
@@ -27,4 +27,4 @@ jobs:
         echo "$HOME/ModelSim-20.1.0/modelsim_ase/bin" >> $GITHUB_PATH
     - name: Test package
       run: |
-        pytest -v -n auto
+        pytest -v -n auto --ignore tests/hdl
```

Comment:
This does not look good. Why you are ignoring some of the tests in CI?

Reply:
Okay, I assumed you thought that you don't have modelsim (though it is automatically installed in CI) and disabled it. I will cherry-pick the rest of your commits, except this one